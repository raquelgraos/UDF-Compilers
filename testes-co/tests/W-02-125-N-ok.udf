public int udf() {
  tensor<3,2,2> t1 = [[[1, 2], [3, 4]],
                      [[5, 6], [7, 8]],
                      [[9, 0], [8, 7]]];
  tensor<2,2,3> t2 = [[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [0, 8, 7]]];
  tensor<3,2,2> t3 = t1 + t1 * t1@(2,1,1) + 3;
  tensor<2,2,3> t4 = t1.reshape(2,2,3);
  writeln 2 * t2;
  writeln t3;
  writeln t1@(2,1,1);
  writeln t4;
  writeln (t1 == t3);
  writeln t1 != t3;
  for (int i = 0; i < t1.rank; i = i + 1) {
    writeln "DIM ", i, " = ", t1.dim(i);
    for (int j = 0; j < t1.dim(i); j = j+1) {
      writeln i, ", ", j;
    }
  }
  writeln sizeof(t1);
}

