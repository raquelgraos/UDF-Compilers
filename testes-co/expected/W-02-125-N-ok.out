Tensor<2,2,3>[[[2, 4, 6], [8, 1E1, 1.2E1]], [[1.4E1, 1.6E1, 1.8E1], [0, 1.6E1, 1.4E1]]]
Tensor<3,2,2>[[[1.1E1, 1.9E1], [2.7E1, 3.5E1]], [[4.3E1, 5.1E1], [5.9E1, 6.7E1]], [[7.5E1, 3], [6.7E1, 5.9E1]]]
7
Tensor<2,2,3>[[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [0, 8, 7]]]
0
1
DIM 0 = 3
0, 0
0, 1
0, 2
DIM 1 = 2
1, 0
1, 1
DIM 2 = 2
2, 0
2, 1
96
